utils/generator-utils.stanza

```diff
@@ -209,14 +209,16 @@ defn connected-group (obj: Pin|Net) :
 ; =======================================
 ; Attach the DNP (do not populate) to objects (components, array of instances, or module)
 ; =======================================
-public defn do-not-populate (objs:Seqable<JITXObject>) :
+public defn dnp (objs:Seqable<JITXObject>) :
   inside pcb-module :
     for obj in objs do :
-      match(instance-type(obj): SingleComponent) : property(obj.DNP) = true
-      else : do-not-populate $ component-instances(obj)
+      switch(instance-type(obj)):
```

Comment:
I have seen `match()` used more than `switch` maybe because it is faster and more robust? 

The function could be modified as follows:
```
match(instance-type(obj)) :
  (o:SingleComponent) : do-not-populate(o)
  (o:SingleModule) : dnp $ instances(o) ; the module could contain sub-modules which component-instance would skip
  (o:InstanceArray) : dnp $ component-instance(o)
```

Reply:
I don't think there is really much of a difference in this case because the `instance-type` in this case isn't returning a value - it is returning a "enumeration" in the form of the types `SingleComponent`, `SingleModule`, and `InstanceArray`. There is an isomorphism between `switch` and `match` in this case. 

However, as you point out, the [component-instances](https://docs.jitx.com/reference/jitx-commands/lookup.html?highlight=component-instances#component-instances) function doesn't recursively find all of the components in the module so that is a bug that I need to fix. That is very non-intuitive.